src/Drips.sol

```diff
@@ -357,23 +357,26 @@ library Drips {
     /// @param defaults The list of default ends, must be sorted by start
     /// @param length The length of the list
     /// @param balance The balance available for drips without duration
-    /// @return end The end time of drips without duration
+    /// @return end_ The end time of drips without duration
     function _receiversDefaultEnd(
         DefaultEnd[] memory defaults,
-        uint256 length,
+        uint8 length,
         uint128 balance
-    ) private pure returns (uint32 end) {
+    ) private pure returns (uint32 end_) {
         uint32 lastStart = 0;
-        uint128 amtPerSec = 0;
-        end = type(uint32).max;
-        for (uint256 i = 0; i < length; i++) {
-            uint32 start = defaults[i].start;
+        uint136 amtPerSec = 0;
+        uint136 end = type(uint136).max;
+        for (uint8 i = 0; i < length; i++) {
+            DefaultEnd memory defaultEnd = defaults[i];
+            uint32 start = defaultEnd.start;
             if (start >= end) break;
-            balance -= amtPerSec * (start - lastStart);
+            balance -= uint128(amtPerSec * (start - lastStart));
             lastStart = start;
-            amtPerSec += defaults[i].amtPerSec;
-            end = _capTimestamp(start + (balance / amtPerSec));
+            amtPerSec += defaultEnd.amtPerSec;
+            end = start + (balance / amtPerSec);
         }
+        if (end > type(uint32).max) end = type(uint32).max;
```

Comment:
Same here

Reply:
The helper casting to `uint32` by capping would be used only once, here, and it'd probably make the logic **harder** to follow.